test/Microsoft.Azure.Devices.Edge.Test.Common/linux/EdgeDaemon.cs

```diff
@@ -17,37 +17,49 @@ public class EdgeDaemon : IEdgeDaemon
 
         public static async Task<EdgeDaemon> CreateAsync(CancellationToken token)
         {
-            string[] platformInfo = await Process.RunAsync("lsb_release", "-sir", token);
-            if (platformInfo.Length == 1)
+            SupportedPackageExtension packageExtension;
+            string os, version;
+            string[] platformInfo = await Process.RunAsync("cat", @"/etc/os-release", token);
+            os = Array.Find(platformInfo, element => element.StartsWith("ID="));
+            version = Array.Find(platformInfo, element => element.StartsWith("VERSION_ID="));
+
+            // VERSION_ID is desired but it is an optional field
+            if (version == null)
             {
-                platformInfo = platformInfo[0].Split(' ');
+                version = Array.Find(platformInfo, element => element.StartsWith("VERSION="));
```

Comment:
The `VERSION` field can have any of the following values:
```
VERSION="18.04.4 LTS (Bionic Beaver)"
VERSION="1.0.20201124"  
VERSION="7 (Core)"
```
We may want to sterile them first before setting the C# `version`

Reply:
Done in new revision